### PR TITLE
Fix imports and auto-import source directories from libraries

### PR DESCRIPTION
**Pull request by mariusgrigoriu**
First, the fix:
Package name is appended in front of the import strings provided by users. But if your BUILD file is located in the workspace root, then the import directory looks like this `/import_dir`

It seems that concatenating workspace_root/package_name/import_dir will address the different scenarios that might pop up. Skipping concatenation when there is no workspace_root or package name prevents adding leading slashes. For example, the examples directory is a workspace_root, but does not have a package name.

Next, auto-import:
I have a workspace where a .jsonnet in root package depends on a generated json from a sub package. The location of the generated file is `bazel-out/.../...` but it doesn't seem right to add `imports = "bazel-out/darwin-fastbuild/..."` so I took a shot at adding the directories for any files. Not sure if this is desired behavior, but I don't know of an alternative. This seems to work well in my project.

### CODE REVIEW

File: jsonnet/jsonnet.bzl [hunk 1]
```diff
@@ -68,7 +77,7 @@ def _jsonnet_library_impl(ctx):
   """Implementation of the jsonnet_library rule."""
   depinfo = _setup_deps(ctx.attr.deps)
   sources = depinfo.transitive_sources + ctx.files.srcs
-  imports = depinfo.imports + ctx.attr.imports
+  imports = depinfo.imports + ctx.attr.imports + [src.dirname for src in ctx.files.srcs]
```
Comment (sparkprime) [inline jsonnet/jsonnet.bzl]: If this adds a -J for every source file mentioned in the jsonnet_library srcs attribute, then that is probably dangerous.  You'll end up with file collisions and weird behavior.

Comment (sparkprime) [reply jsonnet/jsonnet.bzl]: I wonder if you could just have some import_root=True that would at least add `bazel-out/darwin-fastbuild/path/to/rule/dir`.  So then the paths you import in Jsonnet look like the ones in the srcs attribute of the library rule.

Comment (mariusgrigoriu) [reply jsonnet/jsonnet.bzl]: That sounds reasonable. Is there a way to query what that root would be?

Comment (sparkprime) [reply jsonnet/jsonnet.bzl]: I don't actually know anything about Bazel I'm afraid :)

Comment (sparkprime) [reply jsonnet/jsonnet.bzl]: @davidzchen knows things!

Comment (sparkprime) [reply jsonnet/jsonnet.bzl]: For context, see last paragraph of PR description

**Comment by seh** [comment]
Can we split this PR and land commit 12222d209ad96fa1325dc6b6034280040c68f354 to fix #55? I see that @sparkprime [has concerns with the number of `-J` arguments that would be built up](https://github.com/bazelbuild/rules_jsonnet/pull/46#discussion_r183216411).

@sparkprime, can you elaborate on what you meant by this?
> You'll end up with file collisions and weird behavior.

I was hoping that these rules would automatically add a `-J` argument for each of the Jsonnet dependencies, so that if one of my Jsonnet sources imports _lib/stuff.libsonnet_, and I've expressed a dependency on "//lib:stuff" via the "deps" attribute, I shouldn't have to _also_ express explicitly that I wish to include _lib_ as a search directory via `-J` with the "imports" attribute.

**Comment by mariusgrigoriu** [comment]
Closing in favor of #59 which just has the fix when package name is blank. I believe I misinterpreted how rules_jsonnet should be used, and now believe that automatically adding import paths is likely to be undesirable. Instead do something like this `import 'full_path_to_file` where the path is relative to the workspace root.

**Comment by seh** [comment]
@mariusgrigoriu, can you clarify why automatically adding import paths would be a problem? Does it have something to do with accidentally importing more files than intended, or accidentally importing the wrong file due to a name collision?

**Comment by seh** [comment]
I think we're talking about different things here.

Consider [your sample file _examples/BUILD_](https://github.com/mariusgrigoriu/rules_jsonnet/blob/380eb54b13a0ce3c5ee54914c928c1193a229c1d/examples/BUILD) in #59. At line [124,](https://github.com/mariusgrigoriu/rules_jsonnet/blob/380eb54b13a0ce3c5ee54914c928c1193a229c1d/examples/BUILD#L124) you write the following:
```json
imports = ["importdir"]
```
My question is why it's necessary to repeat mention of the _importdir_ directory when [the "import_lib" library](https://github.com/mariusgrigoriu/rules_jsonnet/blob/380eb54b13a0ce3c5ee54914c928c1193a229c1d/examples/BUILD#L135-L138) already specifies that [it's rooted at _importdir_](https://github.com/mariusgrigoriu/rules_jsonnet/blob/380eb54b13a0ce3c5ee54914c928c1193a229c1d/examples/BUILD#L137).
```json
srcs = ["importdir/import_me.jsonnet"]
```
Why repeat that detail?

With a more complex library that spanned multiple directories, I could understand that some of those directories might be "private"—referenced by relative paths to their files from the top-level directory—and we probably don't want all of those on the import path. Perhaps the `jsonnet_library` rule should have a way to declare what directories are required to be imported to use it, for cases where the top-level directories of its "srcs" attribute entries aren't the correct choice.

**Pull request closed without merging**